cmd/kubeadm/app/phases/upgrade/staticpods.go

```diff
@@ -127,6 +127,12 @@ func (spm *KubeStaticPodPathManager) BackupEtcdDir() string {
 }
 
 func upgradeComponent(component string, waiter apiclient.Waiter, pathMgr StaticPodPathManager, cfg *kubeadmapi.MasterConfiguration, beforePodHash string, recoverManifests map[string]string) error {
```

Comment:
nit: change the `cfg` parameter to just `nodeName string` as that is the only thing used in `cfg`
I'd prefer if you parameterized the rollback function to call here instead of baking in this `recoverEtcd` logic... can you do that please?

Reply:
here is the thing, if a separate function used for etcd manifest rollback then:
1. We just end up duplicating the code
2. componentUpgrade will needs to be changed and checked everytime when roll back is called to see which rollback needs to be called, normal or etcd specific,  I think it will bring more confusion than gain from separating. Please let me know if you still wants me to change it.

Reply:
ok, no need to change it